lib/membership/member.js

```diff
@@ -91,10 +90,14 @@ Member.prototype.evaluateUpdate = function evaluateUpdate(update) {
     if (this._isLocalOverride(update)) {
         // Override intended update. Assert aliveness!
         this.ringpop.stat('increment', 'refuted-update');
-        update = _.defaults({
+        var newIncNumber = this.Date.now();
+        update = {
+            source: this.ringpop.whoami(),
+            sourceIncarnationNumber: newIncNumber,
+            address: this.id,
             status: Member.Status.alive,
-            incarnationNumber: this.Date.now()
-        }, update);
+            incarnationNumber: newIncNumber
+        };
```

Comment:
Since you are gossiping this message downstream, I think it is still reasonable to still base it off of the original one changing the fields you need to change, rather than reconstruct. Will be good for forwards compatibility if we decide to add a new field in the future (similarly like google protocol buffers preserve fields that it doesn't understand when forwarding a message).